### PR TITLE
Enable attribute migrated page

### LINKED ISSUE #10508: Finish Migrating "Catalog > Attributes & Features > Add new / edit attribute Group" page
Part of Symfony migration project

## Status

In order to finish the migration of this back-office page, we need to:

**Todo List**:
- [x] Continue the PR https://github.com/PrestaShop/PrestaShop/pull/31502 (see below)
- [x] The PR above misses one behavior: the ability to export the data presented by the [Grid](https://devdocs.prestashop-project.org/8/development/components/grid/) on the backoffice page. Please implement it.
- [x] Fix broken tests
- [x] Fix attribute group edition
- [x] The [Multistore header](https://github.com/PrestaShop/prestashop-specs/blob/7e820f30347fe69ccd3db5a99ab02ef6f7bbdd54/back-office/multistoregeneralspecs.md#multistore-header-only-from-178) must also be added to the new Symfony page (if it is not already implemented). You can look at other migrated Symfony pages to see how it is built.
- [ ] The bug https://github.com/PrestaShop/PrestaShop/issues/10596 was detected on the legacy page. We want to fix it in the Symfony page that will replace the legacy page.
- [ ] Finally, we need a _last Pull Request_ to remove the legacy backoffice page and enable the Symfony page (see below)

## About PR https://github.com/PrestaShop/PrestaShop/pull/31502

The PR https://github.com/PrestaShop/PrestaShop/pull/31502 aimed to create a new Symfony backoffice  page to replace the legacy page  "Catalog > Attributes & Features > Add new / edit attribute Group". The new Symfony page should follow the [coding standards](https://devdocs.prestashop-project.org/8/development/architecture/migration-guide/) we have for all Symfony pages while providing the same behavior that the legacy page. It aims to create an equivalent page, but powered by Symfony and Twig :) .

Just like other Symfony pages, the application layer is Symfony + Twig, but it uses Commands and Queries to fetch and modify data. Command and Query Handlers are adapters that do rely on legacy.

What you must do for PR https://github.com/PrestaShop/PrestaShop/pull/31502 is to fetch it
- Verify it is correct (for you), if not please fix wrong items you have found
- Have it reviewed (and approved) by committers
- The QA will not test it: they will test the last PR (see below)

## About the last PR

First Todo List items will be done by multiple PRs merged into `develop`. So the work of "create a new Symfony page that provides the same behavior/features as the legacy page" will be carried out BUT it is done on a Symfony page not accessible yet in the backoffice. You can browse it if you know the URL. It's like it's hidden.

In order to "finish" the job, a last PR is needed: this is the PR which _removes legacy files_ and enables the LegacyLinks that will redirect old URLs to the new page. So the new Symfony page replaces the legacy page in the back-office. When this is done, the migration is complete.

It is possible QA will find problems when testing the last PR: behaviors that used to work with the legacy page and are not working / implemented in the Symfony page. Then you must fix them.

Please ask @MatShir if you have questions about the behavior or @matks if you have questions about the code

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Enable attribute migrated page, remove feature flag, clean legacy code
| Type?             | new feature
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | UI tests and CI green, test the new attribute and attribute group page (lists, creation forms, edition forms)
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/8618370446
| Fixed issue or discussion?     | Fixes #10508
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Review by M0rgan01 — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]
🚀 

@PrestaShop/qa-functional Please, do a throughout check and verify:
- That the forms redirect to proper place after saving values. If you save a value, it redirects you to the **value list of the attribute**, not the attribute list. Both when creating a new one and editing an existing value.
- That the rows are clickable like in old versions. Click attribute -> opens value list. Click value -> open edit form.

@jolelievre I see one bug ;-)

Snímek obrazovky 2024-02-06 100339

**Review by boherm — DISMISSED** [review]

**Comment by paulnoelcholot** [comment]
Hello @jolelievre,

The auto test returns an error, can you look at this as well as the bug reported by Hlavtox?

Thanks!

**Review by Progi1984 — DISMISSED** [review]
UI Tests in progress

**Comment by jolelievre** [comment]
Blocked by PR on module faceted search https://github.com/PrestaShop/ps_facetedsearch/pull/979

**Comment by jolelievre** [comment]
Blocked by https://github.com/PrestaShop/ps_facetedsearch/pull/983 and https://github.com/PrestaShop/ps_facetedsearch/pull/981

**Review by M0rgan01 — DISMISSED** [review]

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Comment by Hlavtox** [comment]
@jolelievre @paulnoelcholot Some feedback please. :-)

- If you leave public name empty:
  - Old page: The public_name field is required at least in Čeština (Czech).
  - Migrated: An unexpected error occurred. [PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupConstraintException code 50]: Invalid AttributeGroup localized property "public_name" for language with id "1"
- Attribute list is not clickable, on the old page, it gets you to the list of values.
- Attribute value list is not clickable, on the old page, it gets you to the edit page of that value.
- When you create new attribute value, the page title is correct - `New attribute value`, but when editing it, it says `Editing attribute Red` - I already noted this in my previous comment.
- For some reason, when I open a list of attribute values, it automatically opens "rearrange" mode. It doesn't happen in attribute list.

**Review by florine2623 — COMMENTED** [review]
Hello @jolelievre ,

Tested this PR on `develop`.

I have found a few issues.
## Bug 1
The block Color used to displayed correctly. Now there's one line in the input instead of the chosen color.
**Before:**
Screenshot 2024-03-07 at 14 49 33
**After:**
Screenshot 2024-03-07 at 14 48 57

- [ ] Fixed ?

## Bug 2 

On the Attributes pages, I could chose to Add new attribute or Add a new value. Now I only have one choice.
**Before:**
Screenshot 2024-03-07 at 14 53 36
**After:**
Screenshot 2024-03-07 at 14 53 42

- [ ] Fixed ?

## Bug 3

When I add a new value and change the attribute group, the inputs for that attribute are not displayed. In my case, I have to go to my Color attribute and edit it to add a color manually.
Steps to reproduce :
- Go to BO > Attributes page > View `Paper Type`
- Click Add new value button
- In Attribute group > Change it to `Color`
- The inputs for the color attributes are not automatically displayed :x:

Same case the other way around.
- Go to BO > Attributes page > View `Color`
- Click Add new value button
- In Attribute group > Change it to `Dimension`
- The inputs for the color attributes are not automatically removed :x:

https://github.com/PrestaShop/PrestaShop/assets/16019289/e6524081-c975-4c07-bfb8-fd58b6b82b52

- [ ] Fixed ?
---

## Bug 4

The options `Show SQL Query` and `Export to SQL manager` do not work as expected. The query is different compared to the old page and the export gives a Bad Query.
**Before:**
Screenshot 2024-03-07 at 15 33 12
**After:**
Screenshot 2024-03-07 at 15 31 32

- [ ] Fixed ?
---

## Bug 5

When I edit the Attribute Color to a Dropdown list Attribute type, it won't refresh my page with the right Attribute Type. On old page, it was well removed.
I tried clearing cache from PS and browser.

Steps to reproduce :
- Edit Color Attribute 
- Choose Attribute type = Drop-down list > Save
- Click View Color
- See all the Color column still displayed on the list

https://github.com/PrestaShop/PrestaShop/assets/16019289/9fe825e9-0cbc-44ce-bbf0-367e6c8e5f41

The issue is also present on the Product page. When I create a product with combinations, the attributes Color is displayed with a color, when I deleted it. (this last issue is not related to the PR)
Screenshot 2024-03-07 at 15 52 15

- [ ] Fixed ?


---

## Bug 6

The texture can't be added. The color Black is automatically added.


https://github.com/PrestaShop/PrestaShop/assets/16019289/2618d7b1-c277-4fec-bded-ece4faa7b14d

- [ ] Fixed ?

---
## Bug 7

I have reproduced all of the issues mentioned by @Hlavtox https://github.com/PrestaShop/PrestaShop/pull/35270#pullrequestreview-1864601308
Indeed, when I edit an attribute value, the title of the page is misleading.
In my case, it says `Editing attribute Gray`. It should say `Editing attribute value Gray` or `Editing value Gray`.
There's no reference on the old page, as it only says "Attributes".

Screenshot 2024-03-07 at 15 08 14

- [ ] Fixed ?
---
## Bug 8

If you save a new attribute without Public name, you will have an exception.

Screenshot 2024-03-07 at 15 18 39

- [ ] Fixed ?
---
## Bug 9

Attribute list is not clickable, you have to click on the small View button to get to the list of values. On the old page, you could click on the name AND on the View button.

Same issue with Attributes values. you have to click on the Edit button to edit your value. On the old page, you could click on the name AND on the Edit button.


https://github.com/PrestaShop/PrestaShop/assets/16019289/7fa331e1-671f-4081-8e88-32f233de9c12



- [ ] Fixed ?

## Bug10

For some reason, when I open a list of attribute values, it automatically opens "rearrange" mode. It doesn't happen in attribute list.

Need to adapt the attribut group list

Screenshot 2024-03-07 at 15 58 35

- [ ] Fixed ?
--- 


## Not really a bug but feedbacks

### feedback 1
There's an option to Import your Attributes and attributes values. But in the import of PrestaShop, you have to import `Products` with Attributes in order to get the attributes created. So I think this is misleading and should not be an option. There's no Import type Attributes.

Screenshot 2024-03-07 at 15 24 53

### feedback 2

---

It is not possible to filter by Position anymore. https://github.com/PrestaShop/PrestaShop/issues/11217#issuecomment-661141039
**Before:**
Screenshot 2024-03-07 at 15 57 20

**After:**
Screenshot 2024-03-07 at 15 57 03

@MatShir comment: To order you have to select rearrange

### feedback 3

---

The block URL is misleading. It should not say URL. I'm not sure what it was meant to say 🤔 

Screenshot 2024-03-07 at 16 09 54

Regarding the block Meta Title, I'm not sure how to use it.
I added an element, but in the FO, I don't see it anywhere.

Screenshot 2024-03-07 at 16 13 38

@MatShir comment: You need to insert a path, maybe to check with the content team. It will be a bonus on this PR (not blocking) 

Waiting for your feedback 😃 
Thanks!

File: admin-dev/themes/new-theme/scss/components/_form.scss
Comment (jolelievre) [inline admin-dev/themes/new-theme/scss/components/_form.scss]: I'm not sure this fix should be bound to a special class, if it fails on attribute page there are chances it fails on all others Can you check the other usage in the BO of this component and if they are also not integrated correctly then the fix should be generalized
You can use the `.color-widget` as the class the form group to target the correct element

Also you only changed the height of the input but for the color to take the whole height I think you also need to adapt the padding when I tested your change there was a lot of white space around the color input

Comment (Hlavtox) [reply admin-dev/themes/new-theme/scss/components/_form.scss]: Yep, fixed height is generally a bad idea

Comment (jolelievre) [reply admin-dev/themes/new-theme/scss/components/_form.scss]: Not necessarily, because in this case the input field does contain the proper content it's just shrinked to a small line, probably because of the padding So at least padding needs to be fixed, and after that see if the height also needs an update

By the way I just discovered this issue https://github.com/PrestaShop/PrestaShop/issues/35181 in order return status there seems to be an identified bug on a similar input type So it's a good alternative use case to compare and check how the input can be fixed in the whole BO at once

Comment (SharakPL) [reply admin-dev/themes/new-theme/scss/components/_form.scss]: It's not about height though. Color picker is there and it works but just stopped looking like a color picker in develop branch. Now it's just a borderless lump of color invisible when white.

obraz

**Review by matthieu-rolland — DISMISSED** [review]

File: src/Adapter/AttributeGroup/Validate/AttributeGroupValidator.php
Comment (jolelievre) [inline src/Adapter/AttributeGroup/Validate/AttributeGroupValidator.php]: You can't really do this, the validate method returns a boolean but you don't know for which reason so by adding this constraint associated to "empty name" error you remove the one associated to "invalid name"

To handle both type of invalidate I think you need to check manually the empty constraint and throw the exception manually, then you can use the `validateObjectModelLocalizedProperty` for the other invalidaty

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttributeGroupController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttributeGroupController.php:311]: Adding a constraint here and in the CQRS command is a good idea, but you should also add a constraint on the form type This way the error is caught early and displayed in a form inline error It doesn't mean the check in CQRS is useless, it's just another layer of validation, which you should check in the behat tests by the way

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php:95]: You probably need to add constraints in this form type (and maybe in the `AttributeType` as well?)

Comment (nicosomb) [reply src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php:95]: There is already a constraint for the public name here.

Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php:95]: There is no `NotBlank` constraint which is why you're trying to implement

Comment (Hlavtox) [reply src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php:95]: It would be maybe better if this field was optional and was autofilled with the normal name on save.

Name and public name are the same in most cases, would save some typing...

ping @MatShir @kpodemski

File: composer.lock
Comment (jolelievre) [inline composer.lock:1]: You're not supposed to change all these dependencies You probably ran a too global command like `composer update` That's not how we update our dependencies because we want to master them and avoid introducing some bugs because of major/minor updates

This PR was only supposed to update the faceted search module because it contains the fixes to integrate in the migrated page smoothly But the composer.lock for this module is already up to date on develop and has the latest version So you should revert this modification

TLDR; you can keep the modification on composer.json but composer.lock has no need for modifications

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php:95]: Constraints are still missing on the forms, and you also removed them in the CQRS part meaning nothing checks the empty value any more right?

File: composer.lock
Comment (jolelievre) [inline composer.lock:1]: Still not supposed to udpate all this, besides you have conflicts on this file You should accept modifications from develop and ignore/drop all these ones This should make the conflict resolve easier

**Review by nicosomb — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @jolelievre and @nicosomb ,

Checked all the previously mentioned bugs. None of them are corrected 😢 
The exception is always the same.

The PR is well installed on develop, make assets is done without error.

### Bug 1 :x:

Screenshot 2024-04-02 at 10 17 43

### Bug 2 :x:

Screenshot 2024-04-02 at 10 18 57

### Bug 3 :x:

Screenshot 2024-04-02 at 10 10 09

### Bug 4 :x: 

### Bug 5 :x:

Changing attributes type leads to Exception

https://github.com/PrestaShop/PrestaShop/assets/16019289/698e2d5d-5252-4157-a798-011428666634

### Bug 6 :x:

Adding a value in Color with Texture leads to Exception 

Screenshot 2024-04-02 at 10 27 20

https://github.com/PrestaShop/PrestaShop/assets/16019289/7802d52d-bbbb-4e52-b506-61a48041fab1

### Bug 7 :x:

### Bug 8 :x:

Screenshot 2024-04-02 at 10 31 48

### Bug 9 :x:

Screenshot 2024-04-02 at 10 37 02

### Bug 10 :x:

Thanks!
Waiting for your feedback.

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeGroupType.php]: It's either `DefaultLanguage` on the top element so that it blocks ONLY the default language OR NotBlank in the child element so that it blocks ALL the languages

But not both, did you test your modification?

**Review by jolelievre — COMMENTED** [review]
Thanks @nicosomb 

All good!

**Review by boherm — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by AureRita** [comment]
Hi @jolelievre 

Thank you for your PR, I tested it and it seems to works on the issue 1 and 8 as you can see : 

[recording(183).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/a07ed0cc-f706-43be-83d3-9f540e3d7e05)

Link to the auto test : https://github.com/florine2623/testing_pr/actions/runs/8519002395

Because the auto test is :red_circle: we have to correct this issue before merging this PR 

Thank you

**Comment by nicosomb** [comment]
UI tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/8664056569

**Comment by nicosomb** [comment]
UI tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/8706078972

File: src/Core/Grid/Query/AttributeGroupQueryBuilder.php
Comment (boherm) [inline src/Core/Grid/Query/AttributeGroupQueryBuilder.php]: Maybe, we have too much spaces here?

**Review by boherm — CHANGES_REQUESTED** [review]
LGTM! But I think you have a little issue with spaces somewhere :)

**Review by boherm — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Comment by florine2623** [comment]
Latest auto tests ✅ https://github.com/florine2623/testing_pr/actions/runs/8718616685

**Comment by florine2623** [comment]
FYI @paulnoelcholot 😄

**Comment by Hlavtox** [comment]
@florine2623 From my quick check now, half of the issues are still there...

Bug 2 - not fixed
Bug 5 - not fixed
Bug 6 - not fixed
Bug 9 - not fixed
Bug 10 - not fixed

**Comment by nicosomb** [comment]
We planned to open new PR for the other bugs.

**Comment by jolelievre** [comment]
Don't worry @Hlavtox there's already an issue to track down the remaining bugs left to be fixed https://github.com/PrestaShop/PrestaShop/issues/35591

**Comment by jolelievre** [comment]
If you find some more don't hesitate to enrich this issue

**Pull request merged**
